src/files/modals/Modals.js

```diff
@@ -123,15 +129,46 @@ class Modals extends React.Component {
       case ADD_BY_PATH:
         this.setState({ readyToShow: true })
         break
+      case CLI_TUTOR_MODE:
+        this.setState({ command: this.cliCommand(cliOptions, files) }, () => {
+          this.setState({ readyToShow: true })
+        })
+        break
       default:
         // do nothing
     }
   }
 
-  render () {
-    const { show } = this.props
-    const { readyToShow, link, rename } = this.state
+  cliCommand = (action, files) => {
+    let activeCid = ''
+    let fileName = ''
+    let isPinned = ''
+    let path = ''
+    // @TODO: handle multi-select
+    if (files) {
+      activeCid = files[0].cid
+      fileName = files[0].name
+      isPinned = files[0].pinned
+      path = realMfsPath(files[0].path)
+    }
 
+    switch (action) {
+      case cliCmdKeys.DELETE_FILE_FROM_IPFS:
+        return cliCommandList[action](activeCid)
+      case cliCmdKeys.DOWNLOAD_OBJECT_COMMAND:
+        return cliCommandList[action](activeCid)
+      case cliCmdKeys.RENAME_IPFS_OBJECT:
+        return cliCommandList[action](path, fileName)
+      case cliCmdKeys.PIN_OBJECT:
+        return cliCommandList[action](activeCid, isPinned ? 'rm' : 'add')
```

Comment:
I believe WebUI pins recursively:
```suggestion
        return cliCommandList[action](activeCid, isPinned ? 'rm' : 'add -r')
```

Reply:
@lidel by recursive you mean that if we pin a directory, all the files in the directory and any of it's sub-directories and their files gets pinned?

Reply:
i just confirmed, webui doesn;t add/remove pin recursively
![ipfs-add-pin](https://user-images.githubusercontent.com/10788442/90814073-2c612000-e346-11ea-840f-807d2ffc8f9e.png)
 but please let me know which approach should i follow :)